java/jar/jar.go

```diff
@@ -418,10 +417,17 @@ func (i *Info) parseManifest(ctx context.Context, r io.Reader) error {
 		// This attribute is marked as "Deprecated," but there's nothing that
 		// provides the same information.
 		name = hdr.Get("Implementation-Vendor-Id")
+	case hdr.Get("Extension-Name") != "":
+		// This attribute is used by extensions to define their unique identity.
+		name = hdr.Get("Extension-Name")
+	case hdr.Get("Short-Name") != "":
+		// Attribute found in Jenkins plugins.
+		name = hdr.Get("Short-Name")
 	}
 	for _, key := range []string{
 		"Bundle-Version",
 		"Implementation-Version",
+		"Plugin-Version",
 		"Specification-Version",
```

Comment:
Found an example plugin https://plugins.jenkins.io/ssh-credentials/ where `Specification-Version: 0.0` and `Plugin-Version: 305.v8f4381501156`, and the `Plugin-Version` is the correct one